logstash-core-event-java/src/main/java/com/logstash/Event.java

```diff
@@ -167,10 +170,35 @@ public boolean includes(String reference) {
         }
     }
 
+    public byte[] toBinary() throws IOException {
+        HashMap<String, Map<String, Object>> hashMap = new HashMap<>();
+        hashMap.put("DATA", this.data);
+        hashMap.put("META", this.metadata);
+        return CBOR_MAPPER.writeValueAsBytes(hashMap);
+    }
+
+    public static Event fromBinary(byte[] source) throws IOException {
+        if (source == null || source.length == 0) {
+            return new Event();
+        }
+        Object o = CBOR_MAPPER.readValue(source, HashMap.class);
+        if (o == null) {
+            throw new IOException("The binary source deserialized to null");
+        }
+        try {
+            HashMap<String, Map<String, Object>> hashMap = (HashMap<String, Map<String, Object>>) o;
+            HashMap<String, Object> dataMap = (HashMap<String, Object>) hashMap.get("DATA");
+            dataMap.put(METADATA, hashMap.get("META"));
+            return new Event(dataMap);
+        } catch (ClassCastException e) {
+            throw new IOException("incompatible from binary object type=" + o.getClass().getName() + " , only HashMap is supported");
```

Comment:
seems to be more than just that `HashMap` is the only acceptable object. It must be a very specific HashMap with `DATA` and `META`, maybe catching all these exceptions together makes sense?